src/Adapters/Auth/oauth2.js

```diff
@@ -0,0 +1,231 @@
+ * token introspection endpoint (if implemented by the provider).
+ *
+ * The adapter accepts the following config parameters:
+ *
+ * 1. "tokenIntrospectionEndpointUrl" (string, required)
+ *      The URL of the token introspection endpoint of the OAuth2 provider that
+ *      issued the access token to the client that is to be validated.
+ *
+ * 2. "useridField" (string, optional)
+ *      The name of the field in the token introspection response that contains
+ *      the userid. If specified, it will be used to verify the value of the "id"
+ *      field in the "authData" JSON that is coming from the client.
+ *      This can be the "aud" (i.e. audience), the "sub" (i.e. subject) or the
+ *      "username" field in the introspection response, but since only the
+ *      "active" field is required and all other reponse fields are optional
+ *      in the RFC, it has to be optional in this adapter as well.
+ *      Default: - (undefined)
+ *
+ * 3. "appidField" (string, optional)
+ *      The name of the field in the token introspection response that contains
+ *      the appId of the client. If specified, it will be used to verify it's
+ *      value against the set of appIds in the adapter config. The concept of
+ *      appIds comes from the two major social login providers
+ *      (Google and Facebook). They have not yet implemented the token
+ *      introspection endpoint, but the concept can be valid for any OAuth2
+ *      provider.
+ *      Default: - (undefined)
+ *
+ * 4. "appIds" (array of strings, optional)
+ *      A set of appIds that are used to restrict accepted access tokens based
+ *      on a specific field's value in the token introspection response.
+ *      Default: - (undefined)
+ *
+ * 5. "authorizationHeader" (string, optional)
+ *      The value of the "Authorization" HTTP header in requests sent to the
+ *      introspection endpoint.
+ *      Eg. "Basic dXNlcm5hbWU6cGFzc3dvcmQ="
+ *
+ * 6. "debug" (boolean, optional)
+ *      Enables extensive logging using the "verbose" level.
+ *
+ * The adapter expects requests with the following authData JSON:
+ *
+ * {
+ *   "oauth2": {
+ *     "id": "user's OAuth2 provider-specific id as a string",
+ *     "access_token": "an authorized OAuth2 access token for the user",
+ *   }
+ * }
+ */
+
+import logger from '../../logger';
+const Https = require('https');
+const Parse = require('parse/node').Parse;
+const Path = require('path');
+const Url = require('url');
+const Querystring = require('querystring');
+
+const scriptName = Path.basename(__filename, '.js');
+
+// Returns a promise that fulfills if this user id is valid.
+function validateAuthData(authData, options) {
+  const _logger = _getLogger(validateAuthData.name);
+  const loggingEnabled = _isLoggingEnabled(options);
+  if (loggingEnabled) {
+    _logger.verbose('authData = %s, options = %s',  _objectToString(authData), _objectToString(options));
```

Comment:
this should be debug or trace level (as well as all others).

ALso this may leak sensitive infos into logs which is something we try to avoid at all costs.

Reply:
I've removed all verbose logs, only kept the two lines where an error is logged. None of them can contain sensitive information.